clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -366,8 +383,16 @@ public void close() {
         }
     }
 
+    // visible for testing
+    void invokeCompletedOffsetCommitCallbacks() {
+        for (OffsetCommitCompletion completion : completedOffsetCommits)
+            completion.invoke();
+        completedOffsetCommits.clear();
+    }
+
+    public synchronized void commitOffsetsAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCommitCallback callback) {
+        invokeCompletedOffsetCommitCallbacks();
```

Comment:
Ok, I was wondering because isolating it only to `poll()` then I don't think it needs to be synchronized. Probably not a big deal, but now that we have synchronization it'd be good to minimize the scope of it.